src/Nethermind/Nethermind.Blockchain/Contracts/CallableContract.cs

```diff
@@ -29,18 +30,19 @@ protected CallableContract(ITransactionProcessor transactionProcessor, IAbiEncod
             _transactionProcessor = transactionProcessor ?? throw new ArgumentNullException(nameof(transactionProcessor));
         }
 
-        private byte[] Call(BlockHeader header, string functionName, Transaction transaction) => CallCore(_transactionProcessor, header, functionName, transaction);
+        private byte[] Call(BlockHeader header, string functionName, Transaction transaction, IBlockTracer? blockTracer) => CallCore(_transactionProcessor, header, functionName, transaction, blockTracer);
```

Comment:
I would prefer passing ITxTracer and handling rest outside of call here.
If Tracer is optional, lets add it always as last parameter and default to null.

Reply:
yes, I will make the changes. But this PR is not ready for a review, I meant to create this PR on my fork to see an redundant changes that needed to be cleand up.
regarding the optional paramerter - i was thinking of making it non optional and also add this tracer to the withdrawalProcessor also. We dont trace withdrawals right now.